### PR TITLE
fix(vite): allow overriding vite sourcemap

### LINKED ISSUE #7336: NuxtConfig: `vite.build.sourcemap = "hidden"` won't output source map files.
### Environment

-----------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.10-27709424.bb8942a`
- Nitro Version:    `0.5.2-27709289.11cc5cd`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `vite`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

https://stackblitz.com/edit/github-wtf9vy?file=nuxt.config.ts

### Describe the bug

Build the project by running `yarn build` then run `ls .output/public/_nuxt/` to show the result files: no source map files are generated! 

It was working fine in earlier versions before [3.0.0-rc.10-27709172.ec21019](https://www.npmjs.com/package/nuxt3/v/3.0.0-rc.10-27709172.ec21019)

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves  #7336

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Since we started setting sourcemap per-client, per-server, we are overriding the user's choice if they have configured this manually, which they may wish to do.

This respects the value of `vite.build.sourcemap` if set.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by pi0** [comment]
This might be a breaking change but having multiple sources of trust is probably more dangerous to introduce such cases. Any reason we can't rely on one top level `sourcemap` and flag vite option as never with replacement same as https://github.com/nuxt/framework/pull/7317?

**Comment by danielroe** [comment]
I think issue is that vite allows multiple _kinds_ of sourcemaps. e.g. 'hidden' or 'inline'.

**Comment by pi0** [comment]
I see. It is same for webpack too actually. I was going to suggest in initial refactor to extend boolean to other string values per type. But in this case, for now it can be enabled with top level sourcemap.client (it is a breaking change) and keep using vite config until we can do that. I don't think implicitly auto enabling client sourcemap was right behavior.

File: packages/vite/src/client.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ export async function buildClient (ctx: ViteBuildContext) {
       dedupe: ['vue']
     },
     build: {
-      sourcemap: ctx.nuxt.options.sourcemap.client,
+      sourcemap: ctx.config.build?.sourcemap ?? ctx.nuxt.options.sourcemap.client,
```
[the remark this answers is not in the document]
Comment (danielroe) [reply packages/vite/src/client.ts:54]: Nice solution!

**Review by pi0 — APPROVED** [review]

**Pull request merged**